staging/src/k8s.io/endpointslice/util/controller_utils.go

```diff
@@ -62,7 +61,7 @@ func GetPodServiceMemberships(serviceLister v1listers.ServiceLister, pod *v1.Pod
 			// if the service has a nil selector this means selectors match nothing, not everything.
 			continue
 		}
-		key, err := controller.KeyFunc(service)
+		key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(service)
```

Comment:
Interesting, I think that this is not needed at all, but I leave this just as a comment to not forget and revisit later https://github.com/kubernetes/kubernetes/pull/84280/files#r1247603063, no action required